src/bundle/Resources/public/js/alloyeditor/src/buttons/ez-btn-linkedit.js

```diff
@@ -0,0 +1,359 @@
+
+        editor.focus();
+        editor.eZ.moveCaretToElement(editor, this.state.element);
+        editor.fire('actionPerformed', this);
+    }
+
+    /**
+     * Lifecycle. Renders the row of the button.
+     *
+     * @method renderUDWRow
+     * @return {Object} The content which should be rendered.
+     */
+    renderUDWRow() {
+        return (
+            <div className='ez-ae-edit-link__row'>
+                <div className='ez-ae-edit-link__block ez-ae-edit-link__block--udw'>
+                    <button className='ez-ae-button ez-btn-ae ez-btn-ae--udw btn btn-gray' onClick={this.selectContent.bind(this)}>
+                        Select content
+                    </button>
+                </div>
+                <div className='ez-ae-edit-link__block ez-ae-edit-link__block--separator'>
+                    <span className='ez-ae-edit-link__text'>or</span>
+                </div>
+                <div className='ez-ae-edit-link__block ez-ae-edit-link__block--url'>
+                    <label className='ez-ae-edit-link__label'>Link to</label>
+                    <input className='ae-input ez-ae-edit-link__input'
+                        onChange={this.setHref.bind(this)} onKeyDown={this.handleKeyDown.bind(this)}
+                        placeholder='Type or paste link here'
+                        type='text' value={this.state.linkHref}
+                    ></input>
+                    <button aria-label={AlloyEditor.Strings.clearInput}
+                        className='ez-btn-ae ez-btn-ae--clear-link ae-button ae-icon-remove'
+                        onClick={this.clearLink.bind(this)} title={AlloyEditor.Strings.clear}
+                    ></button>
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Lifecycle. Renders the row of the button.
+     *
+     * @method renderInfoRow
+     * @return {Object} The content which should be rendered.
+     */
+    renderInfoRow() {
+        const target = this.state.linkTarget;
+
+        return (
+            <div className='ez-ae-edit-link__row'>
+                <div className='ez-ae-edit-link__block ez-ae-edit-link__block--title'>
+                    <label className='ez-ae-edit-link__label'>Title</label>
+                    <input type='text'
+                        className='ae-input ez-ae-edit-link__input' onChange={this.setTitle.bind(this)}
+                        value={this.state.linkTitle}
+                    ></input>
+                </div>
+                <div className='ez-ae-edit-link__block ez-ae-edit-link__block--target'>
+                    <span className='ez-ae-edit-link__text'>Open in:</span>
+                    <div className='ez-ae-edit-link__choice'>
+                        <label htmlFor='ez-ae-link-target-same' className='ez-ae-edit-link__label ez-ae-edit-link__label--same-tab'>
+                            <input type='radio' name='target' id='ez-ae-link-target-same'
```

Comment:
I'm a bit confused because of code formatting. Please use apostrophes in strings and double quotes in HTML markup. Also for better readability each attribute should be in a  separate line when there's more than 2 attributes per element.
Another thing to consider is to split this render into a few smaller render functions like: `renderTitle` and `renderTarget` or something like this. I believe it could improve reviewer/developer experience in the future.